llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -546,7 +540,7 @@ MachineTraceMetrics::Ensemble::invalidate(const MachineBasicBlock *BadMBB) {
   if (BadTBI.hasValidHeight()) {
     BadTBI.invalidateHeight();
     WorkList.push_back(BadMBB);
-    do {
+    while (!WorkList.empty()) {
```

Comment:
No reason to change this either, especially when it is clear that the container is non-empty on entry.